projects/ui-components/shared/LfSelectable.ts

```diff
@@ -55,12 +59,13 @@ export class Selectable {
     }
     lastCheckedIdx += list.length;
     if (selected.length > 0) {
-      if (this.callback) {
+      if (this.callback && maxFetchIterations > 0) {
+        --maxFetchIterations;
         const value = await this.callback();
         if (!value || value.length === 0) {
           return;
         }
-        await this.setSelectedValuesAsync(selected, value, lastCheckedIdx);
+        await this.setSelectedNodesAsync(selected, value, maxFetchIterations, lastCheckedIdx);
```

Comment:
add a debug if maxFetchIterations is <=0 so the user can know why it just stopped
May not be obvious the default is 5